package.json

```diff
@@ -26,17 +26,92 @@
     "prepare": "husky install",
     "test-install": "npm run test --workspace @puppeteer-test/installation",
     "test-types": "tsd -t packages/puppeteer",
-    "test:chrome:headful": "npm test -- --test-suite chrome-headful",
-    "test:chrome:new-headless": "npm test -- --test-suite chrome-new-headless",
-    "test:chrome:headless": "npm test -- --test-suite chrome-headless",
-    "test:chrome:bidi": "npm test -- --test-suite chrome-bidi",
-    "test:chrome": "run-s test:chrome:*",
-    "test:firefox:bidi": "npm test -- --test-suite firefox-bidi",
-    "test:firefox:headful": "npm test -- --test-suite firefox-headful",
-    "test:firefox:headless": "npm test -- --test-suite firefox-headless",
-    "test:firefox": "run-s test:firefox:*",
+    "test:chrome:headful": "wireit",
+    "test:chrome:new-headless": "wireit",
+    "test:chrome:headless": "wireit",
+    "test:chrome:bidi": "wireit",
+    "test:chrome": "wireit",
+    "test:firefox:bidi": "wireit",
+    "test:firefox:headful": "wireit",
+    "test:firefox:headless": "wireit",
+    "test:firefox": "wireit",
     "test": "cross-env PUPPETEER_DEFERRED_PROMISE_DEBUG_TIMEOUT=20000 node tools/mochaRunner/lib/main.js"
   },
+  "wireit": {
+    "build": {
+      "dependencies": [
+        "./packages/ng-schematics:build",
+        "./packages/puppeteer-core:build",
+        "./packages/puppeteer:build",
+        "./packages/testserver:build",
+        "./test:build",
+        "./test/installation:build"
+      ]
+    },
+    "build:docs": {
+      "dependencies": [
+        "./packages/puppeteer:build:docs",
+        "./packages/puppeteer-core:build:docs"
+      ]
+    },
+    "test:chrome:headful": {
+      "command": "npm test -- --test-suite chrome-headful",
+      "dependencies": [
+        "./test:build"
+      ]
+    },
+    "test:chrome:headless": {
+      "command": "npm test -- --test-suite chrome-headless",
+      "dependencies": [
+        "./test:build"
```

Comment:
we keep changing whether the test command will build the tests or not. Let's decide for one or another way? If the tests do cause a rebuild now, let's update existing documentation to reflect that (README, contributing guides).

Reply:
The tests can build now because of wireit. Mind if we update the docs in another CL? I think it makes sense to dogfood this for a bit before settling.

Reply:
alternatively, let's keep the test commands to be without dependencies so that no automatic rebuild will happen.

Reply:
summarising the discussion offline: we will try the new commands for some time and update the docs later: